sql/src/main/java/io/crate/planner/statement/SetSessionPlan.java

```diff
@@ -62,15 +67,38 @@ public void executeOrFail(DependencyCarrier executor,
                               RowConsumer consumer,
                               Row params,
                               SubQueryResults subQueryResults) throws Exception {
+
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(plannerContext.transactionContext(),
+                                                                              plannerContext.functions(),
+                                                                              x,
+                                                                              params,
+                                                                              subQueryResults);
+
         SessionContext sessionContext = plannerContext.transactionContext().sessionContext();
-        for (Map.Entry<String, List<Expression>> entry : settings.entrySet()) {
-            SessionSetting<?> sessionSetting = SessionSettingRegistry.SETTINGS.get(entry.getKey());
-            if (sessionSetting == null) {
-                LOGGER.info("SET SESSION STATEMENT WILL BE IGNORED: {}", entry.getKey());
-            } else {
-                sessionSetting.apply(params, entry.getValue(), sessionContext);
-            }
+        Assignment<Symbol> assignment = settings.get(0);
+        String settingName = eval.apply(assignment.columnName()).toString();
+        validateSetting(settingName);
+        SessionSetting<?> sessionSetting = SessionSettingRegistry.SETTINGS.get(settingName);
+        if (sessionSetting == null) {
+            LOGGER.info("SET SESSION STATEMENT WILL BE IGNORED: {}", settingName);
+        } else {
+            sessionSetting.apply(sessionContext, assignment.expressions(), eval);
         }
         consumer.accept(InMemoryBatchIterator.empty(SENTINEL), null);
     }
+
+    @VisibleForTesting
+    static void validateSetting(String settingName) {
+        /*
```

Comment:
SQLActionException: 400 Bad Request 0 ERROR: SQLParseException: Setting 'search_path' is not supported
        at __randomizedtesting.SeedInfo.seed([D084E852F8A07CF3:D270444D632F7C4B]:0)